### PR TITLE
enhancement(http source): Add `binary` encoding support for http source

### PR DESCRIPTION
**Pull request by tombokombo**
Hi, this PR adds binary encoding option for http source. Binary protocols works with vector, but only if there is no new line character (\n) in message. If character appears, messages is split by it into multiple messages, which breaks AVRO protocol and basically all binary protocols. Use case is application -> avro bin message -> fluentd -> kafka, and we want to replace fluentd with vector. Custom vector build with this patch is already running in production, everything seems fine. There is new unit test for it as well.

### CODE REVIEW

File: src/sources/util/body_decoding.rs [hunk 1]
```diff
@@ -62,6 +63,9 @@ pub fn decode_body(body: Bytes, enc: Encoding) -> Result<Vec<Event>, ErrorMessag
                 .map_err(|error| json_error(format!("Error parsing Json: {:?}", error)))?;
             json_parse_array_of_object(parsed_json)
         }
+        Encoding::Binary => std::iter::once(Ok(body))
+            .map(|r| Ok(LogEvent::from(r?).into()))
+            .collect::<Result<_, _>>(),
```
Comment (bruceg) [inline src/sources/util/body_decoding.rs:68]: This looks like a complicated way of producing the result. Is this anything more complicated than the following?
```rust
Ok(vec![LogEvent::from(body).into()])
```

**Review by bruceg — COMMENTED** [review]
This looks good, assuming the tests pass, though I have a question about the critical lines in `decode_body`.

File: src/sources/util/body_decoding.rs (hunk 1 above)
Comment (tombokombo) [reply src/sources/util/body_decoding.rs]: Thanks for simplification, that was basically stolen from text encoding part, already simplified according yours suggestion.

**Review by bruceg — APPROVED** [review]
:+1: :shipit:

**Review by jszwedko — APPROVED** [review]
This seems good to me, thanks @tombokombo !

**Pull request merged**